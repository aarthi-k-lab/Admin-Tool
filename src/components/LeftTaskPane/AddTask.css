.add-task.add-task {
    background-color: var(--dark-lime-green);
    border-radius: 10px;
    width: 35%;
    height: 2rem;
    min-height: 0.25rem;
    padding: 0px;
    margin: 0.75rem 0.5rem 0.5rem 5.5rem;
  }

  .add-task span {
    justify-content: left;
  }
  
  .add-task.add-task:hover {
    background-color: var(--dark-lime-green);
  }
  