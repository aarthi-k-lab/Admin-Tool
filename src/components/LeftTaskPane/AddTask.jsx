import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Tooltip from '@material-ui/core/Tooltip';

const styles = theme => ({
  margin: {
    margin: theme.spacing.unit,
  },
  extendedIcon: {
    marginRight: theme.spacing.unit,
  },
  custom: {
    color: 'black',
    backgroundColor: 'var(--grey-300)',
    borderRadius: '0.15rem',
    minHeight: '25px',
    width: '30px',
    height: '0px',
    '&:hover': {
      backgroundColor: 'var(--grey-300)',
    },
  },
});

const AddTask = ({
  classes, onClick, toolTipPosition, margin,
}) => (
  <Tooltip aria-label="Add Task" placement={toolTipPosition} title="Add Task">
    <Fab
      aria-label="Add"
      className={classes.custom}
      color="secondary"
      onClick={onClick}
      style={margin}
    >
      <AddIcon />
    </Fab>
  </Tooltip>
);

AddTask.defaultProps = {
  margin: {
    'margin-left': '10rem',
  },
  onClick: () => {},
  toolTipPosition: 'bottom',
};

AddTask.propTypes = {
  classes: PropTypes.shape.isRequired,
  margin: PropTypes.shape({
    marginLeft: PropTypes.string,
  }),
  onClick: PropTypes.func,
  toolTipPosition: PropTypes.string,
};

export default withStyles(styles)(AddTask);
