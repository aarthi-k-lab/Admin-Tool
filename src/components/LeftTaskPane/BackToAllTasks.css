.back-to-all-tasks.back-to-all-tasks {
    background-color: var(--dark-lime-green);
    border-radius: 10px;
    width: 35%;
    height: 2rem;
    min-height: 0.25rem;
    padding: 0px;
    margin: 0.75rem 0.5rem 0.5rem 5.5rem;
  }

  .back-to-all-tasks span {
    justify-content: left;
  }
  
  .back-to-all-tasks.back-to-all-tasks:hover {
    background-color: var(--dark-lime-green);
  }
  