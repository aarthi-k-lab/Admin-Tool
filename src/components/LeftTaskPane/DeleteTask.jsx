import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Fab from '@material-ui/core/Fab';
import DeleteOutlined from '@material-ui/icons/DeleteOutlined';
import Tooltip from '@material-ui/core/Tooltip';

const styles = theme => ({
  margin: {
    margin: theme.spacing.unit,
  },
  extendedIcon: {
    marginRight: theme.spacing.unit,
  },
  custom: {
    color: 'black',
    backgroundColor: 'var(--grey-300)',
    borderRadius: '0.15rem',
    minHeight: '25px',
    width: '30px',
    height: '0px',
    '&:hover': {
      backgroundColor: 'var(--grey-300)',
    },
  },
});

const DeleteTask = ({
  classes, onClick, toolTipPosition, margin,
}) => (
  <Tooltip aria-label="Delete Task" placement={toolTipPosition} title="Delete Task">
    <Fab
      aria-label="Delete"
      className={classes.custom}
      color="secondary"
      onClick={onClick}
      style={margin}
    >
      <DeleteOutlined />
    </Fab>
  </Tooltip>
);

DeleteTask.defaultProps = {
  margin: {
    'margin-left': '10rem',
  },
  onClick: () => {},
  toolTipPosition: 'bottom',
};

DeleteTask.propTypes = {
  classes: PropTypes.shape.isRequired,
  margin: PropTypes.shape,
  onClick: PropTypes.func,
  toolTipPosition: PropTypes.string,
};

export default withStyles(styles)(DeleteTask);
