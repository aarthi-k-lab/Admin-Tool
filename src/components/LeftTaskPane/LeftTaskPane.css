.taskpane {
  align-self: flex-start;
  background-color: white;
  display: inline-flex;
  flex-direction: column;
  height: 100%;
  list-style-type: none;
  transition: width 250ms ease-in;
}

.task-pane-controls {
  display: flex;
  justify-content: space-between;
  margin-top: 0.5rem;
}

.task-pane-controls-collapsed {
  justify-content: space-around;
}

.status-select-field-span {
  display: inline-flex;
  margin-left: 1.5rem;
}

.status-select-field-text {
  font-size: 4px !important;
}

.collapse-icon-open:focus {
  outline: none;
}

.collapse-icon-closed:focus {
  outline: none;
}

.stretch-column {
  align-self: stretch;
  display: flex;
  flex-direction: column;
}

.error-indicator {
  color: var(--red-a700);
  margin: auto;
}

.loader {
  margin: auto;
}
