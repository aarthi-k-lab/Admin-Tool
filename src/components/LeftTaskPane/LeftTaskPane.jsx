import React from 'react';
import PropTypes from 'prop-types';
import MenuItem from '@material-ui/core/MenuItem';
import TextField from '@material-ui/core/TextField';
import CircularProgress from '@material-ui/core/CircularProgress';
import ErrorIcon from '@material-ui/icons/Error';
import classNames from 'classnames';
import CollapseIcon from 'components/Tasks/CollapseIcon';
import LeftParentTasks from 'components/Tasks/LeftParentTasks';
import TaskModel from 'lib/PropertyValidation/TaskModel';
import OptionalTaskModel from 'lib/PropertyValidation/OptionalTaskModel';
import OptionalTaskDetails from '../Tasks/OptionalTask/OptionalTaskDetails';
import styles from './LeftTaskPane.css';
import AddTask from '../Tasks/OptionalTask/AddTask';
import ChecklistHistory from '../Checklist/ChecklistHistory';

const ALL = 'All';
const PENDING = 'Pending';
const COMPLETED = 'Completed';
const OPEN = 'open';
const statusMap = {
  All: '',
  Pending: 'in-progress',
  Completed: 'completed',
};

function StatusMenu({ onChange, taskStatus }) {
  return (
    <span styleName="status-select-field-span">
      <TextField
        className="status-select-field"
        id="status-selector"
        InputProps={{
          classes: {
            input: 'status-select-field-text',
          },
          disableUnderline: true,
        }}
        label="Status"
        onChange={onChange}
        select
        value={taskStatus}
      >
        <MenuItem value={ALL}>{ALL}</MenuItem>
        <MenuItem value={PENDING}>{PENDING}</MenuItem>
        <MenuItem value={COMPLETED}>{COMPLETED}</MenuItem>
      </TextField>
    </span>
  );
}


function shouldShowAddTaskButton(optionalTasks) {
  return optionalTasks.length > 0;
}

StatusMenu.defaultProps = {
  onChange: () => {},
  taskStatus: ALL,
};

StatusMenu.propTypes = {
  onChange: PropTypes.func,
  taskStatus: PropTypes.string,
};

class LeftTaskPane extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      tasksStatus: ALL,
      width: props.defaultState === OPEN ? props.openWidth : props.closedWidth,
      isCollapsed: props.defaultState !== OPEN,
      // eslint-disable-next-line react/no-unused-state
    };
    this.handleClick = this.handleClick.bind(this);
    this.handleStatusChange = this.handleStatusChange.bind(this);
    // this.handleClose = this.handleClose.bind(this);
    // this.checklistHistoryMenu = this.checklistHistoryMenu.bind(this);
    // this.handleHistoricalCheclistClick = this.handleHistoricalCheclistClick.bind(this);
    // this.handleChecklistOpen = this.handleChecklistOpen.bind(this);
  }

  handleStatusChange(event) {
    const { storeTaskFilter } = this.props;
    const selectedStatus = event.target.value;
    this.setState({
      tasksStatus: selectedStatus,
    });
    storeTaskFilter(statusMap[selectedStatus]);
  }


  handleClick() {
    const { isCollapsed } = this.state;
    const { openWidth, closedWidth } = this.props;
    this.setState({
      width: isCollapsed ? openWidth : closedWidth,
      isCollapsed: !isCollapsed,
    });
  }


  renderContent() {
    const { tasksStatus, isCollapsed } = this.state;
    const {
      dataLoadStatus,
      handleShowOptionalTasks,
      onSubTaskClick,
      resetDeleteTaskConfirmation,
      selectedTaskId,
      optionalTasks,
      tasks,
      updateChecklist,
      handleShowDeleteTaskConfirmation, shouldDeleteTask,
      historicalCheckListData,
    } = this.props;
    if (dataLoadStatus === 'failed') {
      return (
        <ErrorIcon fontSize="large" styleName="error-indicator" />
      );
    }
    if (dataLoadStatus === 'loading') {
      return (
        <CircularProgress styleName="loader" />
      );
    }
    return (
      <>
        <div styleName={isCollapsed ? 'task-pane-controls task-pane-controls-collapsed' : 'task-pane-controls'}>
          {
            !isCollapsed
              ? (
                <>
                  <StatusMenu
                    onChange={this.handleStatusChange}
                    taskStatus={tasksStatus}
                  />
                  <div styleName="icons">
                    <div styleName="checklist-history-icon">
                      <ChecklistHistory
                        checkListData={historicalCheckListData}
                        margin={{ 'margin-left': '5rem' }}
                      />
                    </div>
                    { shouldShowAddTaskButton(optionalTasks)
                      ? <AddTask onClick={() => handleShowOptionalTasks()} />
                      : <div />
                  }
                  </div>
                </>
              )
              : null
          }
          <span
            onClick={this.handleClick}
            onKeyPress={() => null}
            role="button"
            styleName={isCollapsed ? 'collapse-icon-closed' : 'collapse-icon-open'}
            tabIndex={0}
          >
            <CollapseIcon
              direction={isCollapsed ? 'right' : 'left'}
            />
          </span>
        </div>
        <LeftParentTasks
          handleShowDeleteTaskConfirmation={handleShowDeleteTaskConfirmation}
          handleShowOptionalTasks={handleShowOptionalTasks}
          isCollapsed={isCollapsed}
          onSubTaskClick={onSubTaskClick}
          optionalTasks={optionalTasks}
          resetDeleteTaskConfirmation={resetDeleteTaskConfirmation}
          selectedTaskId={selectedTaskId}
          shouldDeleteTask={shouldDeleteTask}
          tasks={tasks}
          updateChecklist={updateChecklist}
        />
      </>
    );
  }


  render() {
    const { width } = this.state;
    const {
      className, handleShowOptionalTasks, optionalTasks, showOptionalTasks, updateChecklist,
      handleShowDeleteTaskConfirmation, shouldDeleteTask, resetDeleteTaskConfirmation,
    } = this.props;

    return (
      <div className={classNames(className, styles['stretch-column'])}>
        <div
          id="cmod_taskpane"
          style={{ width }}
          styleName="taskpane"
        >
          { showOptionalTasks
            ? (
              <OptionalTaskDetails
                handleShowDeleteTaskConfirmation={handleShowDeleteTaskConfirmation}
                handleShowOptionalTasks={handleShowOptionalTasks}
                resetDeleteTaskConfirmation={resetDeleteTaskConfirmation}
                shouldDeleteTask={shouldDeleteTask}
                tasks={optionalTasks}
                updateChecklist={updateChecklist}
              />
            )
            : this.renderContent() }
        </div>
      </div>
    );
  }
}

LeftTaskPane.propTypes = {
  className: PropTypes.string,
  closedWidth: PropTypes.string,
  dataLoadStatus: PropTypes.string,
  defaultState: PropTypes.string,
  handleShowDeleteTaskConfirmation: PropTypes.func.isRequired,
  handleShowOptionalTasks: PropTypes.func.isRequired,
  historicalCheckListData: PropTypes.arrayOf(Object).isRequired,
  onSubTaskClick: PropTypes.func.isRequired,
  openWidth: PropTypes.string,
  optionalTasks: PropTypes.arrayOf(OptionalTaskModel),
  resetDeleteTaskConfirmation: PropTypes.func.isRequired,
  selectedTaskId: PropTypes.string,
  shouldDeleteTask: PropTypes.bool.isRequired,
  showOptionalTasks: PropTypes.bool.isRequired,
  storeTaskFilter: PropTypes.func.isRequired,
  tasks: PropTypes.arrayOf(TaskModel).isRequired,
  updateChecklist: PropTypes.func.isRequired,
};

LeftTaskPane.defaultProps = {
  className: '',
  closedWidth: '4rem',
  dataLoadStatus: 'completed',
  defaultState: 'open', // or 'closed'
  openWidth: '20rem',
  selectedTaskId: '',
  optionalTasks: [],
};

export default LeftTaskPane;
