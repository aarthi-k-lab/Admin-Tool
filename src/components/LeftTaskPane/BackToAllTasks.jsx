import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import ChervronLeft from '@material-ui/icons/ChevronLeft';
import './BackToAllTasks.css';

const BackToAllTasks = ({ disabled, onClick }) => (
  <Button
    className="material-ui-button"
    color="green"
    disabled={disabled}
    onClick={onClick}
    styleName="back-to-all-tasks"
    variant="contained"
  >
    <ChervronLeft />
    <span>BACK TO ALL TASKS</span>
  </Button>
);

BackToAllTasks.defaultProps = {
  disabled: false,
  onClick: () => {},
};

BackToAllTasks.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
};

export default BackToAllTasks;
