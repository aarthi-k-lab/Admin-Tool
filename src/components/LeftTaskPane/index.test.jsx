import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import LeftTaskPane from './LeftTaskPane';

describe('<LeftTaskPane Components />', () => {
  it('LeftTaskPane renders correctly', () => {
    const tree = renderer
      .create(<LeftTaskPane historicalCheckListData={[]} optionalTasks={[]} tasks={[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
