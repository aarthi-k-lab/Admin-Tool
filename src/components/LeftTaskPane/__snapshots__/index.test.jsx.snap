// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeftTaskPane Components /> LeftTaskPane renders correctly 1`] = `
<div
  className="stretch-column"
>
  <div
    className="components-LeftTaskPane-LeftTaskPane__taskpane"
    id="cmod_taskpane"
    style={
      Object {
        "width": "20rem",
      }
    }
  >
    <div
      className="components-LeftTaskPane-LeftTaskPane__task-pane-controls"
    >
      <span
        className="components-LeftTaskPane-LeftTaskPane__status-select-field-span"
      >
        <div
          className="MuiFormControl-root-1 status-select-field"
        >
          <label
            className="MuiFormLabel-root-16 MuiFormLabel-filled-20 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12"
            data-shrink={true}
            htmlFor="status-selector"
          >
            Status
          </label>
          <div
            className="MuiInputBase-root-43 MuiInput-root-30 MuiInputBase-formControl-44 MuiInput-formControl-31"
            onClick={[Function]}
          >
            <div
              className="MuiSelect-root-23"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                className="MuiSelect-select-24 MuiSelect-selectMenu-27 MuiInputBase-input-53 MuiInput-input-38 status-select-field-text"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                All
              </div>
              <input
                id="status-selector"
                type="hidden"
                value="All"
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-60 MuiSelect-icon-29"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </span>
      <button
        className="MuiButtonBase-root-99 MuiButton-root-73 MuiButton-contained-84 MuiButton-containedPrimary-85 MuiButton-raised-87 MuiButton-raisedPrimary-88 material-ui-button components-LeftTaskPane-AddTask__add-task"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-74"
        >
          <img
            alt="add-icon"
            height="25"
            src="/static/img/add.svg"
            width="25"
          />
          <span>
            ADD TASK
          </span>
        </span>
        <span
          className="MuiTouchRipple-root-102"
        />
      </button>
      <span
        className="components-LeftTaskPane-LeftTaskPane__collapse-icon-open"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-60"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
    </div>
    <div
      className="components-Tasks-LeftParentTasks__left-pane-tasks-main-div"
    >
      <div
        className="components-Tasks-LeftParentTasks__left-pane-scrollable-tasks"
      />
    </div>
  </div>
</div>
`;
